File: EasyTable/Demos/Delphi/MergeDBExe/Unit1.pas
unit Unit1;

interface

uses
  Windows, Messages, SysUtils, Classes, Graphics, Controls, Forms, Dialogs, EasyTable;

type
  TForm1 = class(TForm)
    procedure FormCreate(Sender: TObject);
  private
    { Private declarations }
  public
    { Public declarations }
  end;

var
  Form1: TForm1;

implementation

{$R *.DFM}

procedure TForm1.FormCreate(Sender: TObject);
var ExeFile,DBFile: String;
    Database:       TEasyDatabase;
begin
  ExeFile := ParamStr(1);
  DBFile := ParamStr(2);
  Database := TEasyDatabase.Create(nil);
  try
    Database.DatabaseName := 'edb';
    Database.DatabaseFileName := DBFile;
    Database.MakeExeDatabase(ExeFile,ExeFile+'1');
    DeleteFile(ExeFile);
    RenameFile(ExeFile+'1',ExeFile);
  finally
    Database.Free;
    Close;
    Application.Terminate;
  end;
end;

end.

File: EasyTable/Demos/C++ Builder 5/MergeDBExe/Unit1.cpp
//---------------------------------------------------------------------------
#include <vcl.h>
#include "EasyTable.hpp"
#include <SysUtils.hpp>
#pragma hdrstop

#include "Unit1.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma resource "*.dfm"
TForm1 *Form1;
//---------------------------------------------------------------------------
__fastcall TForm1::TForm1(TComponent* Owner)
        : TForm(Owner)
{
}
//---------------------------------------------------------------------------

void __fastcall TForm1::FormCreate(TObject *Sender)
{
  AnsiString pExeFile=ParamStr(1);
  AnsiString pDBFile=ParamStr(2);
  TEasyDatabase* Database = new TEasyDatabase(NULL);
  try
   {
    Database->DatabaseName = "edb";
    Database->DatabaseFileName = pDBFile;
    Database->MakeExeDatabase(pExeFile,pExeFile+"1");
    DeleteFile(pExeFile);
    RenameFile(pExeFile+"1",pExeFile);
   }
  __finally
   {
    delete Database;
    Close();
    Application->Terminate();
   }
}
//---------------------------------------------------------------------------
